### DISCUSSION TITLE
(IMPORTANT) need help bypassing an error

### DISCUSSION THREAD

**Discussion by discipleofbert** [grame-cncm/faust#993] [Q&A]
ok so i'm writing some code and i get an error that says that "the parameter must be a real constant numerical expression", which bothers me since i want to be able to change the parameter in real time. below is the code; the code causing the error in question is in bold...

```
declare name 		"whammy";
import("stdfaust.lib");

whammy = hgroup("Whammy",  
                                ef.dryWetMixer(
                                vslider("wetAmount[style:knob]", 1, 0, 1, 0.05),
                                  ef.transpose(
									1000, 10, vslider("shift (semitones)", 0, **vslider("lower", 0, -24, 24, 1)**, **vslider("upper", 0, -24, 24, 1)**, 0.01)
								  )

                                )
				);

process = whammy;
```
is there any way i can bypass this error and have it still work as intended? thx

**Comment by DBraun** [comment]
When you make a vslider for "shift (semitones)", the init, min, max, and step must be [compile-time constants](https://faustdoc.grame.fr/manual/syntax/#hslider-primitive). Your code uses vsliders for the min and max, but sliders aren't constant. Does this code satisfy your goal?

```faust
declare name "whammy";
import("stdfaust.lib");

whammy = hgroup("Whammy", ef.dryWetMixer(vslider("wetAmount[style:knob]", 1, 0, 1, 0.05),
    ef.transpose(1000, 10,
        (vslider("shift (semitones)", 0, -24, 24, 0.01) : si.smoo)
    )
));

process = whammy;
```

I put a si.smoo in there too.

**Reply by discipleofbert** [reply]
wait actually i want to be able to change the highest and lowest pitches through the gui
